pkg/ebpf/tracer.go

```diff
@@ -273,13 +274,9 @@ func (t *Tracer) initPerfPolling() (*bpflib.PerfMap, error) {
 				}
 				atomic.AddInt64(&t.perfReceived, 1)
 				cs := decodeRawTCPConn(conn)
-				cs.Direction = t.determineConnectionDirection(&cs)
-				if t.shouldSkipConnection(&cs) {
-					atomic.AddInt64(&t.skippedConns, 1)
-				} else {
-					cs.IPTranslation = t.conntracker.GetTranslationForConn(cs.Source, cs.SPort, process.ConnectionType(cs.Type))
-					t.state.StoreClosedConnection(cs)
-				}
+				cs.IPTranslation = t.conntracker.GetTranslationForConn(cs.Source, cs.SPort, process.ConnectionType(cs.Type))
+				// Store all connections, we will determine if we should skip/if it is closed in GetConnections
```

Comment:
Why? Do we know how many more connections we'll be storing in state? I've seen 10k+ local dns connections before, which I'd prefer to leave out of the state...

Reply:
I wasn't aware that we were using direction to skip connections, when was that added?

Reply:
This was moved because we now need the full list of connections to determine connection direction and we only have one available here.

shouldSkipConnection is defined as

```
func (t *Tracer) shouldSkipConnection(conn *ConnectionStats) bool {
	isDNSConnection := conn.DPort == 53 || conn.SPort == 53
	if !t.config.CollectLocalDNS && isDNSConnection && conn.Direction == LOCAL {
		return true
	} else if IsBlacklistedConnection(t.sourceExcludes, t.destExcludes, conn) {
		return true
	}
	return false
}
```

The only place it uses Local is to enfore the older CollectLocalDNS option which was replaced by connection blacklisting.  How about removing the `conn.Direction == LOCAL` check and going back to calling `shouldSkipConnection` before a connection gets stored in `t.state`